libvast/src/system/meta_index.cpp

```diff
@@ -446,7 +455,11 @@ meta_index(meta_index_actor::stateful_pointer<meta_index_state> self,
                  metrics_metadata{{"query", id_str}});
       self->send(self->state.accountant, "meta-index.lookup.candidates",
                  result.size(), metrics_metadata{{"query", std::move(id_str)}});
-      return result;
+      // TODO: This reproduces the previous behavior, but it is a pessimization:
```

Comment:
The reader of this code won't necessarily know what the previous behavior was. Please expand the comment.